docs: refactor Monitoring and add exporting metrics (#790)
GreptimeTeam/docs: docs/v0.6/en/user-guide/operations/configuration.md

## Configuration File

### Examples

Configurations can be used in one or multiple components according to their features.
You can find all available configurations for each component on GitHub:

- [standalone](https://github.com/GreptimeTeam/greptimedb/blob/main/config/standalone.example.toml)
- [frontend](https://github.com/GreptimeTeam/greptimedb/blob/main/config/frontend.example.toml)
- [datanode](https://github.com/GreptimeTeam/greptimedb/blob/main/config/datanode.example.toml)
- [metasrv](https://github.com/GreptimeTeam/greptimedb/blob/main/config/metasrv.example.toml)

### Specify configuration file

You can specify the configuration file by using the command line arg `-c [file_path]`.

```sh
greptime [standalone | frontend | datanode | metasrv]  start -c config/standalone.example.toml
```

> For example, start the standalone mode as below:

Reviewer: ```suggestion
For example, start the standalone as below:
```

```bash
greptime standalone start -c standalone.example.toml
```

### Protocol options

> Protocol options are valid in `frontend` and `standalone` sub command, which specify the protocol server address and other protocol-related options.

Reviewer: ```suggestion
Protocol options are valid in `frontend` and `standalone` sub commands, which specify the protocol server addresses and other protocol-related options.
```

Below is an example configuration with default values. 
The HTTP and gRPC protocols must be enabled for the database to work correctly.
The other protocols are optional.

> If you want to disable certain options, such as OpenTSDB protocol support, you can set the `enable` parameter to `false`.

Reviewer: ```suggestion
If you want to disable certain protocols, such as OpenTSDB protocol support, you can set the `enable` parameter to `false`.
```

|            | addr         | String  | Server address, "127.0.0.1:4003" by default                                     |
|            | runtime_size | Integer | The number of server worker threads, 2 by default                               |

> <!-- #### Node options

Reviewer: We still need this section,  because there is a `node_id` for datanode.

Author: `node_id` option is in the `Datanode-only configuration` section.

There are also some node options in common:

| Option | Key  | Type   | Description                                                |
| ------ | ---- | ------ | ---------------------------------------------------------- |

> |        | mode | String | Node running mode, includes "standalone" and "distributed" | -->

Reviewer: Keep it before we remove it.

Author: Are there any default values here?
Default values make users do not need to care about the options.

- `purge_threshold` and `purge_interval`: control the purging of wal files, default is `4GB`.
- `sync_write`: whether to call `fsync` when writing every log.

### Logging options

`frontend`, `metasrv`, `datanode` and `standalone` can all configure log and tracing related parameters in the `[logging]` section:

```toml
[logging]
dir = "/tmp/greptimedb/logs"
level = "info"
enable_otlp_tracing = false
otlp_endpoint = "localhost:4317"
tracing_sample_ratio = 1.0
append_stdout = true
```

- `dir`: log output directory.
- `level`: output log level, available log level are `info`, `debug`, `error`, `warn`, the default level is `info`.

> - `enable_otlp_tracing`: whether to turn on distributed tracing, not turned on by default.

Reviewer: ```suggestion
- `enable_otlp_tracing`: whether to turn on tracing, not turned on by default.
```

```toml
[export_metrics]
# Whether to enable export_metrics
enable=true
# Export time interval
write_interval = "30s"
[export_metrics.self_import]
db = "information_schema"
```

- `db`: The default database used by `self_import` is `information_schema`. You can also create another database for saving system metrics.

#### `remote_write` method

The `remote_write` method is supported by `datanode`, `frontend`, `metasrv`, and `standalone`.
It sends metrics to a receiver compatible with the [Prometheus RemoteWrite protocol](https://prometheus.io/docs/concepts/remote_write_spec/).

```toml
[export_metrics]
# Whether to enable export_metrics
enable=true
# Export time interval
write_interval = "30s"
[export_metrics.remote_write]
# URL specified by Prometheus RemoteWrite protocol
url = "http://127.0.0.1:4000/v1/prometheus/write?db=information_schema"
# Some optional HTTP parameters, such as authentication information
headers = { Authorization = "Basic Z3JlcHRpbWVfdXNlcjpncmVwdGltZV9wd2Q=" }
```

- `url`: URL specified by Prometheus RemoteWrite protocol.
- `headers`: Some optional HTTP parameters, such as authentication information.

> ### Mode option

Author: @killme2008 PTAL
